juju/juju
# [JUJU-2897] back ssh allow list with model config

This is a step towards supporting ssh/controller ingress rules configuration. 

Remove the firewall rules api from the client, as a step towards dropping the facade from the controller

At the moment, this model-config item is not implemented, so behaviour will be the same as prior to this PR

## Checklist


## QA steps

Bootstrap a controller, add a model and verify ssh-allowlist has the correct default
```
$ juju bootstrap lxd lxd
$ juju add-mdoel m
$ juju model-config ssh-allowlist
0.0.0.0/0
$ juju list-firewall-rules
WARNING This command now just sets/reads the "ssh-allowlist" model-config item and
is deprecated in favour of setting/reading that item with "juju model-config"

Service  Whitelist subnets
ssh      0.0.0.0/0
```

Set ssh-allowlist using model-config
```
$ juju model-config ssh-allowlist="192.168.0.0/24"
$ juju model-config ssh-allowlist
192.168.0.0/24"
$ juju list-firewall-rules
WARNING This command now just sets/reads the "ssh-allowlist" model-config item and
is deprecated in favour of setting/reading that item with "juju model-config"

Service  Whitelist subnets
ssh      192.168.0.0/24
```

Using set-firewall-rule
```
$ juju set-firewall-rule ssh --allowlist "192.168.1.0/24"
WARNING This command now just sets/reads the "ssh-allowlist" model-config item and
is deprecated in favour of setting/reading that item with "juju model-config"
$ juju model-config ssh-allowlist
192.168.1.0/24"
$ juju list-firewall-rules
WARNING This command now just sets/reads the "ssh-allowlist" model-config item and
is deprecated in favour of setting/reading that item with "juju model-config"

Service  Whitelist subnets
ssh      192.168.1.0/24
```

#### Also,
Verify the `cmr` integration test passes to verify firewall changes
```
./main.sh -v -p aws -c aws cmr
```
